src/main.cpp

```diff
@@ -104,10 +104,34 @@ createCacheDirectory()
 int
 main(int argc, char *argv[])
 {
-        // needed for settings so need to register before any settings are read to prevent warings
+        // needed for settings so need to register before any settings are read to prevent warnings
         qRegisterMetaType<UserSettings::Presence>();
 
-        QCoreApplication::setApplicationName("nheko");
+        // This is some hacky programming, but it's necessary (AFAIK?) to get the unique config name parsed
+        // before the app name is set.
+        QString appName{"nheko"};
+        for (int i = 0; i < argc; ++i)
+        {
+                if (QString{argv[i]}.startsWith("--profile="))
+                {
+                        QString q{argv[i]};
+                        q.remove("--profile=");
+                        appName = q;
```

Comment:
I think this will cause issues, if someone sets their profile name to mozilla, kde, gnome or some other application name. I think it is safer to append in this case?